### PR TITLE
feat(helpdesk) CHK-1865 implemented business logic for retrieve user wallet by fiscal code or email

### PR DESCRIPTION
**Pull request by Fernando-Granato**
#### List of Changes
Created two new endpoints to perform a search for a user's active payment methods.
The search can be performed by fiscal code or user email

#### Motivation and Context
The change was made to provide assistance with an api to retrieve a user's payment methods

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/PaymentMethodDataProvider.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import it.pagopa.generated.ecommerce.helpdesk.model.HelpDeskSearchTransactionRequestDto
+import it.pagopa.generated.ecommerce.helpdesk.model.PmSearchPaymentMethodRequestDto
+import it.pagopa.generated.ecommerce.helpdesk.model.SearchPaymentMethodResponseDto
+import reactor.core.publisher.Mono
+
+/**
+ * Payment method data provider interface. This interface models common function for payment method
+ * data provider in order to search for payment method given a specific
+ * HelpDeskSearchTransactionRequestDto criteria
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/PaymentMethodDataProvider.kt:11]:
```suggestion
 * PmSearchPaymentMethodRequestDto criteria
```

File: src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/PaymentMethodDataProvider.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * @see HelpDeskSearchTransactionRequestDto
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/PaymentMethodDataProvider.kt:13]:
```suggestion
 * @see PmSearchPaymentMethodRequestDto
```

File: src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/oracle/PmQueries.kt [hunk 3]
```diff
@@ -151,3 +151,83 @@ val userFiscalCodeCountQuery =
...
+		ps.ID_SATISPAY AS FK_SATISPAY,
+		pw.FK_BPAY, pb.BANK_NAME AS BPAY_NAME ,pb.CELLPHONE_NUMBER AS BPAY_NUMBER,
+		pw.FK_GENERIC_INSTRUMENT, pgi.DESCRIPTION AS GENERIC_INSTRUMENT_DESCRIPTION,
+		pp2.ID  AS FK_PPAL, pp2.EMAIL_PP AS PPAL_EMAIL 
+        FROM AGID_USER.PP_USER pu 
+        left JOIN AGID_USER.PP_WALLET pw ON pu.ID_USER =pw.ID_USER 
+        left  JOIN AGID_USER.PP_CREDIT_CARD pcc  ON pw.FK_CREDIT_CARD  = pcc.ID_CREDIT_CARD 
+        left JOIN AGID_USER.PP_MYBANK_BUYER_BANK pmbb  ON pw.FK_BUYER_BANK  = pmbb.ID_BUYER_BANK 
+        left JOIN AGID_USER.PP_BANCOMAT_CARD pbc  ON pw.FK_BANCOMAT_CARD  = pbc.ID_BANCOMAT_CARD 
+        left JOIN AGID_USER.PP_SATISPAY ps  ON pw.FK_SATISPAY = ps.ID_SATISPAY AND pw.FK_SATISPAY IS NOT null --pw.ID_USER  = ps.ID_USER and  pw."TYPE"  ='3'
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/oracle/PmQueries.kt:182]:
```suggestion
        left JOIN AGID_USER.PP_SATISPAY ps  ON pw.FK_SATISPAY = ps.ID_SATISPAY AND pw.FK_SATISPAY IS NOT null
```
remove this query comment

**Review by pietro-tota — DISMISSED** [review]
Well done! Just some trivial issues

**Review by infantesimone — DISMISSED** [review]

File: src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/PaymentMethodDataProvider.kt (hunk 1 above)
Comment (Fernando-Granato) [reply src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/PaymentMethodDataProvider.kt]: Fixed with b84c18840faedb2d0d4c1e609f9688ec20501a96

File: src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/PaymentMethodDataProvider.kt (hunk 2 above)
Comment (Fernando-Granato) [reply src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/PaymentMethodDataProvider.kt]: Fixed with b84c18840faedb2d0d4c1e609f9688ec20501a96

File: src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/oracle/PmQueries.kt (hunk 3 above)
Comment (Fernando-Granato) [reply src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/oracle/PmQueries.kt]: Fixed with 23580a36a8e5d732bc23936bbd4a78d39a7e3cfb

File: api-spec/openapi.yaml [hunk 4]
```diff
@@ -558,6 +657,191 @@ components:
           PAYMENT_TOKEN: "#/components/schemas/SearchTransactionRequestPaymentToken"
           TRANSACTION_ID: "#/components/schemas/SearchTransactionRequestTransactionId"
           USER_EMAIL: "#/components/schemas/SearchTransactionRequestEmail"
+    PaypalDetailInfo:
```
[the remark this answers is not in the document]
Comment (Fernando-Granato) [reply api-spec/openapi.yaml:660]: Fixed with b408a08edadeec3ccabca305c301a1ac81326fc6

**Review by pietro-tota — APPROVED** [review]
🚀

**Pull request merged**